File: MATTHEW/leetcode/AIcoding/5.Qwen-3/javascript_solutions/506-relative-ranks.js
function findRelativeRanks(score) {
    const sorted = [...score].sort((a, b) => b - a);
    const rankMap = new Map();
    for (let i = 0; i < sorted.length; i++) {
        if (i === 0) rankMap.set(sorted[i], "Gold Medal");
        else if (i === 1) rankMap.set(sorted[i], "Silver Medal");
        else if (i === 2) rankMap.set(sorted[i], "Bronze Medal");
        else rankMap.set(sorted[i], (i + 1).toString());
    }
    return score.map(s => rankMap.get(s));
}

File: leetcode/AIcoding/5.Qwen-3/cpp_solutions/506-relative-ranks.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& nums) {
        vector<int> sorted = nums;
        sort(sorted.begin(), sorted.end(), greater<int>());
        map<int, int> rankMap;
        for (int i = 0; i < sorted.size(); ++i) {
            rankMap[sorted[i]] = i + 1;
        }
        vector<string> result;
        for (int num : nums) {
            int rank = rankMap[num];
            if (rank == 1) {
                result.push_back("Gold Medal");
            } else if (rank == 2) {
                result.push_back("Silver Medal");
            } else if (rank == 3) {
                result.push_back("Bronze Medal");
            } else {
                result.push_back(to_string(rank));
            }
        }
        return result;
    }
};